### ISSUE TITLE
compatible versions of R

### ISSUE THREAD

**Issue by mesamuels** [toshi-ara/brunnermunzel#15]
I tried installing this package on our system (Digital Research Alliance of Canada, previously Compute Canada), from the CRAN site. I got an error message that the package was incompatible with that version of R (v 4.2.2). I tried a couple of other versions we have and got the same error message. Rather than keep trying randomly, can someone clarify what versions of R are compatible with this package (Brunner Munzel)? Thanks in advance!

**Comment by toshi-ara** [comment]
I can install this package into R-4.2.2 and R-4.3.0 (Windows).
Also, I can install this into R-4.2.* and R-4.3.0 in Linux (Debian 11).
However, I have never seen  incompatible error. 

In Linux (or Mac?), gfotran is required.

**Comment by mesamuels** [comment]
Thanks for the quick response. I presume you mean 'gfortran' not 'gfotran'. I'm not sure exactly what that means in terms of actual commands I need to run, but I will ask our system support group, I imagine they are familiar with this situation. It's not urgent as it turns out we already have Brunner-Munzel in our Python scipy-stack, but it's still a useful exercise for me to learn in case it comes up again with something else r-specific. Cheers from Montreal!

**Comment by toshi-ara** [comment]
"gfortran" is correct.
This  package requires gcc and gfortran to compile from source code.

I hope you succeed install this package.

**Comment by mesamuels** [comment]
I understand, however it is confusing in terms of the package contents. I downloaded the file brunnermunzel_2.0.tar.gz to my Unix account, and did the various unzipping and unarchiving to get the directory brunnermunzel. It has some files including DESCRIPTION, MDS, NAMESPACE, etc., as well as the subdirectories build, inst, man, R, src, tests, vignettes. There is no obvious README file or instructions on exactly how to compile. In looking through the subdirectories, I don't see any files with a .f extension typical for fortran code. Can you explain how to actually do the install on a unix system? Or do I have the wrong file. When I tried running the link to CRAN.R-project.org/package=brunnermunzel](https://cran.r-project.org/package=brunnermunzel , directly from my unix command prompt that's when I got the incompatibility error. Even if I have to ask for help from my system support (Compute Canada), they don't know the details of every package they install, so some further guidance would be helpful. As to the version I already have on the system via Python, I don't think it includes the Brunner-Munzel permutation test, which I should also run as my sample size is small, just a few more than 10.

Thanks,
Mark

**Comment by toshi-ara** [comment]
To install package,
`install.packages("brunnermunzel")` in R console (preferable)
or
`R CMD INSTALL brunnermunzel_2.0.tar.gz` at file-existing folder in terminal.

I have not tried to install to "Unix system". However, this package can be installed in Mac system (https://cran.r-project.org/web/checks/check_results_brunnermunzel.html).

Fortran codes (.f) are present in src folder.

**Comment by mesamuels** [comment]
Thanks I will try this, or at worst pass this info along to the support group (FYI, Compute Canada has reformed as the Digital Research Alliance (DRA) of Canada, in case you ever get contacted from them. Cheers from Montreal!

**Comment by mesamuels** [comment]
OK I was able to install using the R CMD INSTALL command from the unix prompt. When I tried the install.packages command from the R prompt, it asked for a CRAN mirror site, which I didn't know. It is now installed in a directory R. I am having trouble loading it in R, it says there is no package brunnermunzel. Does the package have another name? Exactly what directory do I need to be in when I start R, or what is the directory level since once installed there are four levels in my new directory named R in my home directory: /myhomedir/R/x86_64-pc-linux-gnu-library/4.2/brunnermunzel . Apologies, I am new to R and this is the first time I am trying to run a program I have installed locally, versus already available on the system.

**Comment by mesamuels** [comment]
Aha I figured it out. I installed the package in a new subdirectory called rlib, in case the program did not have readwrite access to my home directory. Then from inside R I could load the package with the command: library("brunnermunzel", lib.local = "/home/samuelsm/rlib")

Then from the R prompt I can run brunnermunzel.test. I have to figure out how to use the permutations, but at least I can run it. Thanks!
